packages/react-async/src/component.tsx

```diff
@@ -64,7 +64,7 @@ export function createAsyncComponent<
         load={load}
         id={id}
         defer={defer}
-        renderLoading={renderLoading}
+        renderLoading={() => renderLoading && renderLoading(componentProps)}
```

Comment:
I don't think we really want to return undefined here if `renderLoading` is not provided

Reply:
We also probably want to make it so that the `async.renderLoading` prop is called with the props, too.